utils.js

```diff
@@ -1,15 +1,24 @@
 const { not, equals, pick } = require('ramda')
 
-async function createTable({ dynamodb, name, attributeDefinitions, keySchema }) {
+async function createTable({ dynamodb, name, attributeDefinitions, keySchema, stream }) {
   const res = await dynamodb
     .createTable({
       TableName: name,
       AttributeDefinitions: attributeDefinitions,
       KeySchema: keySchema,
-      BillingMode: 'PAY_PER_REQUEST'
+      BillingMode: 'PAY_PER_REQUEST',
+      ...(stream && {
+        StreamSpecification: {
+          StreamEnabled: true,
+          StreamViewType: 'NEW_IMAGE'
+        }
```

Comment:
I'm not following this snippet. Could you explain this bit of JS magic here? 😊

Reply:
@eahefnawy If you're referring to the `...(stream && {` bit, that is a conditional object spread. So `StreamSpecification` will only be appended to the object if `stream` is truthy.

Reply:
I also don't see `StreamEnabled` as a property of the [CF StreamSpecification configuration](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-streamspecification.html).